service/controller/resource/prometheusautoscaler/resource.go

```diff
@@ -0,0 +1,106 @@
+package prometheusautoscaler
+
+import (
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	autoscaling "k8s.io/api/autoscaling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	vpa_types "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	vpa_clientset "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/client/clientset/versioned"
+
+	"github.com/giantswarm/prometheus-meta-operator/service/controller/resource/generic"
+	"github.com/giantswarm/prometheus-meta-operator/service/key"
+)
+
+const (
+	Name = "prometheusautoscaler"
+)
+
+type Config struct {
+	VpaClient vpa_clientset.Interface
+	Logger    micrologger.Logger
+}
+
+func New(config Config) (*generic.Resource, error) {
+	clientFunc := func(namespace string) generic.Interface {
+		c := config.VpaClient.AutoscalingV1().VerticalPodAutoscalers(namespace)
+		return wrappedClient{client: c}
+	}
+
+	c := generic.Config{
+		ClientFunc:       clientFunc,
+		Logger:           config.Logger,
+		Name:             Name,
+		GetObjectMeta:    getObjectMeta,
+		GetDesiredObject: getObject,
+		HasChangedFunc:   hasChanged,
+	}
+	r, err := generic.New(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return r, nil
+}
+
+func getObjectMeta(v interface{}) (metav1.ObjectMeta, error) {
+	cluster, err := key.ToCluster(v)
+	if err != nil {
+		return metav1.ObjectMeta{}, microerror.Mask(err)
+	}
+
+	return metav1.ObjectMeta{
+		Name:      "prometheus",
+		Namespace: key.Namespace(cluster),
+		Labels:    key.Labels(cluster),
+	}, nil
+}
+
+func getObject(v interface{}) (metav1.Object, error) {
```

Comment:
Can we rename it to toVerticalPodAutoscaler as we usually do?

Reply:
It feels easier to have `getObject` everywhere.

Reply:
Then let's align on this

Reply:
Yeah, the `to*` functions in some other resources are from before we split and renamed the struct members to `GetObjectMeta` and `GetDesiredObject`.